app/search/gleaner.py

```diff
@@ -263,6 +265,25 @@ def text_search(self, **kwargs):
         self.query = GleanerSearch.build_query(user_query, None, page_number)
         return self.execute_query(page_number)
 
+    # A helper method to escape char for different matches
+    def escape_char(self,text):
+        count = text.count('"') #for " " (double quotes)
+        
+
+        #if it has a double quotes and the count of the double quotes is odd add an extra quote e.g "Biobasis " Zackenberg" (double quotes)
+        if count%2==1 :
+            text = text.replace('"','\\"')
+            text = text + '\\"'
+
+        #If the texts has even count of double quotes escape them e.g "Biobasis  Zackenberg" (double quotes)
+        elif count%2==0:
+            
+            text = text.replace('"','\\"')
```

Comment:
Looking good! To make this even more efficient, you could put `text = text.replace('"','\\"')` before the first `if`, and then leave the `elif` out entirely.

Reply:
Yeah that is more efficient

Reply:
hmm, when i do that it seems to fail the test and give an error on the UI when i search something like "Biobasis " Zackenberg"
![image](https://user-images.githubusercontent.com/45835101/196454079-e3989945-6a33-480f-bc03-52e0100335d5.png)
```suggestion
            text = text.replace('"','\\"')
```

Reply:
Huh, that's so strange. Well, no harm in keeping it the way it is.